addons/mail/tests/test_mail_gateway.py

```diff
@@ -366,6 +510,51 @@ def test_message_parse(self):
         self.assertEqual(res['body'], '')
         self.assertEqual(res['attachments'][0][0], 'thetruth.pdf')
 
+    @mute_logger('odoo.addons.mail.models.mail_thread')
+    def test_message_parse_eml(self):
+        """ Test parsing of mail with embedded emails as eml(msg) files.
+        """
+
+        # Tree structure of the message:
+        def display(message_as_tree):
+            return d(message_as_tree, "", "")
```

Comment:
Please avoid nested methods with names that do not mean anything.

Reply:
Also if all "display" does is call d with two parameters set to a default value, why not just use actual default values on the main function?

Reply:
Because technically d is a private method of display, that I didn't nest.
Also this writing style is common because you generally can't avoid it if you want to write tail recursive functions.

Reply:
I'm reading that `d` is useless. This is a helper function of a test module, it can just sit at the toplevel of the module and take two optional parameters. And a docstring or better naming to explain what it actually does.